### PR TITLE
feat(core): 13203 add cookie banner to disaster ninja

### PR DESCRIPTION
**Pull request by Akiyamka**

### CODE REVIEW

File: src/core/cookie_settings/CookieManagementService.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+      case permissionStrategy.prompt:
+        return permissionStatuses.promptNeeded;
+    }
+  }
+
+  /**
+   * Strategy it kind of "fallback behavior"
+   * For example - when user choose "Apply all"
+   * This service will allow to use any cookies for feature requests
```
Comment (Komzpa) [inline src/core/cookie_settings/CookieManagementService.ts:40]: @katluk is that permitted?

**Comment by propakov** [comment]
see this just on app start 
image
looks like` this.ym(this.yandexAccountId, name, payload);` should be wrapped with some `if`

File: src/App.tsx [hunk 2]
```diff
@@ -3,28 +3,16 @@ import { reatomContext } from '@reatom/react';
 import { StrictMode } from 'react';
 import { store } from '~core/store/store';
 import { AuthWrapper } from '~core/auth';
-import { runGoogleTagManager } from '~utils/metrics/tagManager';
 import { Router } from '~core/router';
-
+import { CookieConsentBanner } from '~features/cookie_consent_banner';
 const root = createRoot(document.getElementById('root')!);
```
Comment (propakov) [inline src/App.tsx]:
```suggestion
import { CookieConsentBanner } from '~features/cookie_consent_banner';

const root = createRoot(document.getElementById('root')!);
```

File: src/core/cookie_settings/PermissionStatusWatcher.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export class PermissionStatusWatcher {
+  private listeners = new Set<PermissionStatusListener>();
+  private permission: Permission;
+
+  constructor(permission: Permission) {
+    this.permission = permission;
+  }
+
+  onStatusChange(statusChangeCb: (status: CookiePermissionStatus) => void) {
```
Comment (propakov) [inline src/core/cookie_settings/PermissionStatusWatcher.ts]:
```suggestion
  onStatusChange(statusChangeCb: PermissionStatusListener) {
```

File: src/App.tsx [hunk 4]
```diff
@@ -3,28 +3,16 @@ import { reatomContext } from '@reatom/react';
...
 root.render(
   <StrictMode>
     <reatomContext.Provider value={store}>
       <AuthWrapper>
         <Router />
+        <CookieConsentBanner />
```
Comment (propakov) [inline src/App.tsx]: hm, can't we move it to other common features place in src/views/CommonView.tsx ? 🤔 
not sure it's critical, but I would expect it there

File: src/core/cookie_settings/CookieManagementService.ts (hunk 1 above)
Comment (katluk) [reply src/core/cookie_settings/CookieManagementService.ts:40]: Yes, for now it is ok. During the next stage we are going to introduce the mechanism to manage cookies.

**Comment by Akiyamka** [comment]
> see this just on app start image looks like` this.ym(this.yandexAccountId, name, payload);` should be wrapped with some `if`

@propakov the interesting thing that it have if. So that's mean that somehow ym was deactivated?
  I can't reproduce it, are you remember steps?

image

**Comment by propakov** [comment]
@Akiyamka 
sure
on login page
1 Application -> Clean site data
2 Log in
3 Screenshot 2022-12-19 at 16 59 51
It stays after page refresh

**Comment by propakov** [comment]
@Akiyamka i've debugged it, it runs mark method, but doesn't run init in this case

File: src/features/cookie_consent_banner/components/CookieConsentBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { useAtom } from '@reatom/react';
+import ReactMarkdown from 'react-markdown';
+import { userResourceAtom } from '~core/auth';
+import { AppFeature } from '~core/auth/types';
+import { i18n } from '~core/localization';
+import { cookieSettingsAtom } from '../atoms/cookieSettingsAtom';
+import s from './CookieConsentBanner.module.css';
+
+export function CookieConsentBanner() {
+  const [havePrompts, { acceptAll, rejectAll }] = useAtom(cookieSettingsAtom);
```
Comment (propakov) [inline src/features/cookie_consent_banner/components/CookieConsentBanner.tsx:11]: shouldn't this atom run in forseRun and Banner be like ?

```
      <Suspense fallback={null}>
        {userModel?.hasFeature(AppFeature. USE_3RDPARTY_ANALYTICS) && < CookieConsentBanner />}
      </Suspense>
```

File: src/App.tsx (hunk 4 above)
Comment (Akiyamka) [reply src/App.tsx]: fixed

File: src/features/cookie_consent_banner/components/CookieConsentBanner.tsx (hunk 5 above)
Comment (Akiyamka) [reply src/features/cookie_consent_banner/components/CookieConsentBanner.tsx:11]: @propakov thank you, fixedl

**Review by propakov — APPROVED** [review]
🔥

**Pull request merged**